plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -594,6 +597,10 @@ public void testCreateTableAsWithSchemaLocation()
     @Test
     public void testCreateTableWithSchemaLocation()
     {
+        if (!hasBehavior(SUPPORTS_INSERT)) {
```

Comment:
might want to add a test method called `verifySupportsInsert` to assert that INSERTs actually fail as expected for subclasses which don't support INSERTs.